### PR TITLE
add possibility to override namespace of spirv-cross

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **sprv-cross/any**

it is needed for creation of dirigent package
https://github.com/conan-io/conan-center-index/issues/1533#issuecomment-872322674

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/spirv-cross/all/conanfile.py [hunk 1]
```diff
@@ -95,6 +97,7 @@ def _configure_cmake(self):
         self._cmake.definitions["SPIRV_CROSS_SKIP_INSTALL"] = False
         self._cmake.definitions["SPIRV_CROSS_FORCE_PIC"] = self.options.get_safe("fPIC", True)
         self._cmake.configure(build_folder=self._build_subfolder)
+        self._cmake.definitions["SPIRV_CROSS_NAMESPACE_OVERRIDE"] = self.options.namespace
```
Comment (SpaceIm) [inline recipes/spirv-cross/all/conanfile.py:100]:
```suggestion
        self._cmake.definitions["SPIRV_CROSS_NAMESPACE_OVERRIDE"] = self.options.namespace
        self._cmake.configure(build_folder=self._build_subfolder)
```

**Review by jgsogo — COMMENTED** [review]
Do we want `SPIRV_CROSS_NAMESPACE_OVERRIDE` to always take a value or should we leave it unset if no explicit value is given to the option?

```python
   "namespace": ["ANY", None]

   // default value
   "namespace": None
...

if self.options.namespace:
    self._cmake.definitions["SPIRV_CROSS_NAMESPACE_OVERRIDE"] = self.options.namespace
```

I'm not familiar with the source code of the library and I've never used it, so I cannot enumerate pros and cons.

**Comment by SpaceIm** [comment]
@jgsogo When this definition is not set, namespace is `spirv_cross`.

**Comment by jgsogo** [comment]
> @jgsogo When this definition is not defined, namespace is `spirv_cross`.

I know, I've seen it in the sources, but I don't know if there is some if/ifdef clause that will be executing some extra code if there is a custom namespace. It shouldn't, but who knows 🤷

**Comment by AndreyMlashkin** [comment]
Another thing is that it makes possible to link spirv-cross library multiple times with this override without any conflict, but as far as I know, conan does not allow that. Even with private=True

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
CCI recipes can not consume requirements with none default options 🤔 I am not sure this will help with the library you require to package

**Review by SSE4 — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Pull request merged**